src/App.tsx

```diff
@@ -1,24 +1,166 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
+import { FilterValues, USER_ID } from './constants';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { UserWarning } from './components/UserWarning';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [errorType, setErrorType] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodoAdding, setIsTodoAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState(FilterValues.ALL);
+  const [removingTodoIds, setRemovingTodoIds] = useState<number[]>([]);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (selectedFilter) {
+        case FilterValues.COMPLETED:
+          return todo.completed;
+
+        case FilterValues.ACTIVE:
+          return !todo.completed;
+
+        case FilterValues.ALL:
+          return true;
+
+        default: return todo;
+      }
+    });
+  }, [selectedFilter, todos]);
+
+  const hasActive = todos.some(todoItem => !todoItem.completed);
+
+  const getTodosFromServer = async () => {
+    try {
+      const response = await getTodos(USER_ID);
+
+      setTodos(response);
+    } catch (error) {
+      setErrorType('upload');
+    }
+  };
+
+  const addTodoToServer = useCallback(async () => {
+    const newTodoToFetch = {
+      title,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    const newTodoToShow = {
+      ...newTodoToFetch,
+      id: 0,
+    };
+
+    setIsTodoAdding(true);
+    setTempTodo(newTodoToShow);
+
+    try {
+      const addedTodo = await addTodo(newTodoToFetch);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch (error) {
+      setErrorType('add');
+    } finally {
+      setIsTodoAdding(false);
+      setTempTodo(null);
+    }
+  }, [title]);
```

Comment:
```suggestion
  }, []);
```
pass `title` as an argument - anyway it's available in the Header component too (in this case function will not be created again after each title change)

Reply:
moreover - it's better to have `title` state in the Header component in this case